flows/deindex.py

```diff
@@ -0,0 +1,449 @@
+    cleanups_task_batches = iterate_batch(
+        data=documents_batches, batch_size=cleanups_task_batch_size
+    )
+
+    batches_with_failures: set[int] = set()
+
+    for i, cleanups_task_batch in enumerate(cleanups_task_batches, start=1):
+        logger.info(f"Processing clean-ups task batch #{i}")
+
+        cleanups_tasks = [
+            cleanup_objects_for_batch_flow_or_deployment(
+                documents_batch=documents_batch,
+                documents_batch_num=documents_batch_num,
+                cache_bucket=cache_bucket,
+                concepts_counts_prefix=concepts_counts_prefix,
+                aws_env=aws_env,
+                as_deployment=as_deployment,
+            )
+            for documents_batch_num, documents_batch in enumerate(
+                cleanups_task_batch, start=1
+            )
+        ]
+
+        logger.info(f"Gathering cleanups tasks for batch #{i}")
+        results: list[None | FlowRun] = await asyncio.gather(
+            *cleanups_tasks, return_exceptions=True
+        )
+        logger.info(f"Gathered cleanups tasks for batch #{i}")
+
+        for result in results:
+            if isinstance(result, Exception):
+                batches_with_failures.add(i)
+                logger.error(
+                    f"result was an exception. Message: `{result}`",
+                )
+                break
+
+            if not as_deployment:
+                if result is None:
+                    continue
+            else:
+                # Explicitly narrow the type to FlowRun
+                if not isinstance(result, FlowRun):
+                    batches_with_failures.add(i)
+                    logger.error(
+                        f"result was not one of the expected types: {type(result)}"
+                    )
+                    continue
+
+                # At this point, result is definitely a FlowRun
+                flow_run: FlowRun = result
+
+                if flow_run.state is None:
+                    batches_with_failures.add(i)
+                    logger.error(
+                        f"flow run result's state was unexpectedly missing. Flow run name: `{flow_run.name}`",
+                    )
+                    continue
+                else:
+                    if flow_run.state.type != StateType.COMPLETED:
+                        batches_with_failures.add(i)
+                        logger.error(
+                            f"flow run result's state was not completed. Flow run name: `{flow_run.name}`",
+                        )
+
+    if len(batches_with_failures) > 0:
+        failed_batches_str = ", ".join(str(b) for b in batches_with_failures)
+        raise ValueError(
+            f"Failed to process document batch in clean-ups task batch. Batches with failures: #{failed_batches_str}"
+        )
+
+
+@flow(
+    on_failure=[SlackNotify.message],
+    on_crashed=[SlackNotify.message],
+)
+async def deindex_labelled_passages_from_s3_to_vespa(
```

Comment:
@jesse-c just to confirm my understanding of this, so we do the following: 

- remove all the old labelled passages and concept counts for a classifier in s3 
- run de-indexing into vespa clearing down the concept on the labelled passages that were tagged by the current primary version / latest 

And thus, post run we should have only the "primary" or latest labelled passages in s3?

Reply:
Depending on the classifier specs. parameter (e.g. `Q770:latest`, yes. If it's `latest` for the alias (AKA version), then it will do a big wipe. This would just be triggered off of a concept being removed from the concept store.

Otherwise, if it's as specific alias (e.g. `Q770:v4`), it will remove the artifacts in S3 and the spans from Vespa. We don't know what version is in Vespa, so we can't remove a span _iff_ it's not the latest. The indexing pipeline though does overwrites, so getting the primary (AKA latest) in is expected to be done through there.

De-indexing really is just getting data out, and not getting it up-to-date with latest inference results or classifier specs.